### PR TITLE
ResDiff Integration: Switch to Modulus Distributed Util

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

**Comment by mnabian** [comment]
Verified `test.sh` runs with these changes.

**Review by daviddpruitt — APPROVED** [review]

File: modulus/experimental/resdiff/generate.py [hunk 1]
```diff
@@ -329,8 +330,17 @@ def main(max_times: Optional[int], seeds: List[int], **kwargs):
     
     opts = dnnlib.EasyDict(kwargs)
     
-    dist.init()
-    
+    # Initialize distributed manager.
+    DistributedManager.initialize()
+
+    # wrapper class for distributed manager for print0. This will be removed when Modulus logging is implemented.
+    class DistributedManagerWrapper(DistributedManager):
```
[the remark this answers is not in the document]
Comment (mnabian) [reply modulus/experimental/resdiff/generate.py:337]: This is a workaround until the Modulus logger is used.

File: modulus/experimental/resdiff/train.py [hunk 2]
```diff
@@ -110,8 +117,17 @@ def main(**kwargs):
...
     """
     opts = dnnlib.EasyDict(kwargs)
-    torch.multiprocessing.set_start_method('spawn', force=True)
-    dist.init()
+
+    # Initialize distributed manager.
+    DistributedManager.initialize()
+
+    # wrapper class for distributed manager for print0. This will be removed when Modulus logging is implemented.
+    class DistributedManagerWrapper(DistributedManager):
```
[the remark this answers is not in the document]
Comment (mnabian) [reply modulus/experimental/resdiff/train.py:125]: Yes, I'll remove the wrapper once I have the Modulus logger in the code.

File: modulus/experimental/resdiff/generate.py [hunk 3]
```diff
@@ -213,10 +214,10 @@ def randint(self, *args, size, **kwargs):
         return torch.stack([torch.randint(*args, size=size[1:], generator=gen, **kwargs) for gen in self.generators])
 
 
-def load_pickle(network_pkl):
+def load_pickle(network_pkl, dist):
```
Comment (NickGeneva) [inline modulus/experimental/resdiff/generate.py:217]: something to consider is instead of making this a parameters, just create dist in side the method everytime. Up to you, but the singleton allows you to keep the API clean.

File: modulus/experimental/resdiff/training/dataset.py [hunk 4]
```diff
@@ -283,10 +282,11 @@ def _load_raw_labels(self):
 
 #Era5
 class Era5Dataset(torch.utils.data.Dataset):
-  def __init__(self, params, path, train, cache=True, task='sr'):
+  def __init__(self, params, path, train, dist, cache=True, task='sr'):
```
Comment (NickGeneva) [inline modulus/experimental/resdiff/training/dataset.py:285]: For constructors I would definitely suggest just doing instead of argument.

self.dist = DistributedManager()

File: modulus/experimental/resdiff/training/loss.py [hunk 5]
```diff
@@ -204,14 +203,14 @@ def __init__(self, P_mean=0.0, P_std=1.2, sigma_data=0.5):
         self.P_std = P_std
         self.sigma_data = sigma_data
         
-        print('dist.get_rank', dist.get_rank)
-        if dist.get_rank == 0:
-            breakpoint()
+        # print('dist.get_rank', dist.get_rank)
```
Comment (NickGeneva) [inline modulus/experimental/resdiff/training/loss.py:206]: Is this suppose to be still commented out? Just checking.

File: modulus/experimental/resdiff/training/loss.py [hunk 6]
```diff
@@ -204,14 +203,14 @@ def __init__(self, P_mean=0.0, P_std=1.2, sigma_data=0.5):
...
+        # if dist.get_rank == 0:
+        #     breakpoint()
          
         with torch.no_grad():
             #resume_state_dump = '/lustre/fsw/nvresearch/mmardani/output/logs/edm_era5_cwb_12chans_fcn_4x_crop448_zarr_fulldata_unetregression/complex-mayfly_2023.04.25_16.42/output/training-state-006924.pt'
             resume_state_dump = '/lustre/fsw/nvresearch/mmardani/output/logs/edm_era5_cwb_12chans_fcn_4x_crop448_zarr_fulldata_unetregression/mindful-parakeet_2023.04.03_22.54/output/training-state-042650.pt'
-            dist.print0(f'Loading training state from "{resume_state_dump}"...')
+            print(f'Loading training state from "{resume_state_dump}"...')
```
Comment (NickGeneva) [inline modulus/experimental/resdiff/training/loss.py:213]: Will this print for every rank?

**Review by NickGeneva — APPROVED** [review]

File: modulus/experimental/resdiff/generate.py (hunk 3 above)
Comment (Alexey-Kamenev) [reply modulus/experimental/resdiff/generate.py:217]: Another suggestion is to lift the check up (out of `load_pickle`) and have something like:
```py
if dist.rank == 0:
    load_pickle(pkl1)
    load_pickle(pkl2)
   ...
```

**Review by Alexey-Kamenev — APPROVED** [review]
LGTM.

File: modulus/experimental/resdiff/generate.py (hunk 3 above)
Comment (mnabian) [reply modulus/experimental/resdiff/generate.py:217]: Thanks for the comment. I'm now instantiating the `dist` object inside the method

File: modulus/experimental/resdiff/training/loss.py (hunk 5 above)
Comment (mnabian) [reply modulus/experimental/resdiff/training/loss.py:206]: This is debugging code so it's safe t comment it out

File: modulus/experimental/resdiff/training/loss.py (hunk 6 above)
Comment (mnabian) [reply modulus/experimental/resdiff/training/loss.py:213]: Yes, but once the logger is implemented I'll fix this to print on rank zero only
